pkgs/development/compilers/ghc/8.6.5.nix

```diff
@@ -10,7 +10,7 @@
 , # GHC can be built with system libffi or a bundled one.
   libffi ? null
 
-, useLLVM ? !stdenv.targetPlatform.isx86
+, useLLVM ? !stdenv.targetPlatform.isx86 || (stdenv.targetPlatform.isMusl && stdenv.hostPlatform != stdenv.targetPlatform) || stdenv.targetPlatform.isiOS
```

Comment:
It looks like you just copied this line from the `8.4.4.nix`, but if you know why `stdenv.targetPlatform.isMusl && stdenv.hostPlatform != stdenv.targetPlatform` is needed, it'd be really great to have a comment here.

Why is LLVM needed only when we're cross-compiling to musl?

Reply:
That was an oversight from rebasing, remove it now.